src/PointwiseFunctions/AnalyticData/GrMhd/CcsnCollapse.cpp

```diff
@@ -0,0 +1,909 @@
+
+  return {equation_of_state_.specific_internal_energy_from_density(
+      rest_mass_density)};
+}
+
+// Electron fraction
+template <typename DataType>
+tuples::TaggedTuple<hydro::Tags::ElectronFraction<DataType>>
+CcsnCollapse::variables(
+    const gsl::not_null<IntermediateVariables<DataType>*> vars,
+    const tnsr::I<DataType, 3>& /*x*/,
+    tmpl::list<hydro::Tags::ElectronFraction<DataType>> /*meta*/) const {
+  return {Scalar<DataType>{vars->electron_fraction.value()}};
+}
+
+// Once temperature is added to hydro tags for the grmhd system,
+// temperature will be stored similar to below
+// // Temperature
+// template <typename DataType>
+// tuples::TaggedTuple<hydro::Tags::Temperature<DataType>>
+// CcsnCollapse::variables(
+//     const gsl::not_null<IntermediateVariables<DataType>*> vars,
+//     const tnsr::I<DataType, 3>& /*x*/,
+//     tmpl::list<hydro::Tags::Temperature<DataType>> /*meta*/) const {
+//   return {Scalar<DataType>{vars->temperature.value()}};
+// }
+
+// Velocity (typical spherical coord system rotating around z axis)
+template <typename DataType>
+tuples::TaggedTuple<hydro::Tags::SpatialVelocity<DataType, 3>>
+CcsnCollapse::variables(
+    const gsl::not_null<IntermediateVariables<DataType>*> vars,
+    const tnsr::I<DataType, 3>& x,
+    tmpl::list<hydro::Tags::SpatialVelocity<DataType, 3>> /*meta*/) const {
+  interpolate_vars_if_necessary(vars);
+  auto spatial_velocity = make_with_value<tnsr::I<DataType, 3>>(x, 0.0);
+  // x and y components will have a radial velocity (from file),
+  // and toroidal velocity (from rotation law) component.
+
+  using std::cos;
+  using std::sin;
+  // Omega = omega0/(1 + (r/A)^2)
+  // vphi = distance_from_rotation_axis * Omega
+  auto rotational_velocity =
+      vars->radius * vars->sin_theta * central_angular_velocity_ /
+      (1.0 + square(vars->radius * inv_diff_rot_parameter_));
+
+  // vx = vr*sin(theta)*cos(phi) - vphi*sin(phi)
+  get<0>(spatial_velocity) =
+      vars->fluid_velocity.value() * vars->sin_theta * cos(vars->phi) -
+      rotational_velocity * sin(vars->phi);
+
+  // vy = vr*sin(theta)*sin(phi) + vphi*cos(phi)
+  get<1>(spatial_velocity) =
+      vars->fluid_velocity.value() * vars->sin_theta * sin(vars->phi) +
+      rotational_velocity * cos(vars->phi);
+
+  // vz = vr*cos(theta)
+  get<2>(spatial_velocity) =
+      vars->fluid_velocity.value() * sqrt(1.0 - square(vars->sin_theta));
+  return {std::move(spatial_velocity)};
+}
+
+// Lorentz
+template <typename DataType>
+tuples::TaggedTuple<hydro::Tags::LorentzFactor<DataType>>
+CcsnCollapse::variables(
+    const gsl::not_null<IntermediateVariables<DataType>*> vars,
+    const tnsr::I<DataType, 3>& x,
+    tmpl::list<hydro::Tags::LorentzFactor<DataType>> /*meta*/) const {
+  interpolate_vars_if_necessary(vars);
+
+  auto lorentz_factor = make_with_value<Scalar<DataType>>(get<0>(x), 0.0);
+  // Compute spatial proper velocity, then Lorentz factor.
+  const auto spatial_metric =
```

Comment:
just FYI, here there are allocations too, but we have figured out a good way to fix this (yet) with analytic solutions/data, and it's a low priority since it's _only_ in the analytic prescriptions. So leave as-is :)